libs/model/src/community/GetMembers.query.ts

```diff
@@ -0,0 +1,264 @@
+        node.url,
+      );
+      // add balances to profiles
+      for (const profile of allCommunityProfiles) {
+        for (const address of profile.addresses) {
+          profile.stake_balances ||= [];
+          profile.stake_balances.push(balances[address] || '0');
+        }
+      }
+    }
+
+    const paginatedResults = allCommunityProfiles
+      .slice()
+      .sort((a, b) => {
+        let comparison = 0;
+        switch (payload.order_by) {
+          case 'created_at':
+            comparison = moment(a.created_at).isAfter(b.created_at) ? 1 : -1;
+            break;
+          case 'profile_name':
+            comparison = a.profile_name.localeCompare(b.profile_name);
+            break;
+          default:
+            comparison = moment(a.last_active).isAfter(b.last_active) ? 1 : -1;
+        }
+        return payload.order_direction === 'DESC' ? -comparison : comparison;
+      })
+      .slice(
+        (payload.cursor - 1) * payload.limit,
+        payload.cursor * payload.limit,
+      );
+
+    const profilesWithAddresses = paginatedResults.map((profile) => {
+      return {
+        id: profile.id,
+        user_id: profile.user_id,
+        profile_name: profile.profile_name,
+        avatar_url: profile.avatar_url,
+        addresses: profile.address_ids.map((_, i) => {
+          const address: any = {
+            id: profile.address_ids[i],
+            community_id: profile.community_ids[i],
+            address: profile.addresses[i],
+          };
+          if (profile.stake_balances) {
+            address.stake_balance = profile.stake_balances?.[i] || '0';
+          }
+          return address;
+        }),
+        roles: [] as string[],
+        group_ids: [] as number[],
+      };
+    });
+
+    if (payload.include_roles) {
+      const profileAddressIds = uniq(
+        profilesWithAddresses
+          .map((p) => p.addresses.map((address) => address.id))
+          .flat(),
+      );
+
+      const addressesWithRoles = await models.Address.findAll({
+        where: {
+          id: {
+            [Op.in]: profileAddressIds,
+          },
+          community_id: payload.community_id,
+        },
+      });
+
+      const addressIdRoles: Record<number, string> = addressesWithRoles.reduce(
+        (acc, address) => {
+          return {
+            ...acc,
+            [`${address.id!}`]: address.role,
+          };
+        },
+        {},
+      );
+
+      // add roles to associated profiles in response
+      for (const profile of profilesWithAddresses) {
+        for (const address of profile.addresses) {
+          profile.roles ||= [];
+          profile.roles.push(addressIdRoles[address.id]);
+        }
+      }
+    }
+
+    if (payload.include_group_ids) {
+      const profileAddressIds = uniq(
+        profilesWithAddresses
+          .map((p) => p.addresses.map((address) => address.id))
+          .flat(),
+      );
+
+      const existingMemberships = await models.Membership.findAll({
```

Comment:
can we include this in a single query? ... similar to loading aggregates, queries should be a single call, unless it's a composite query reading from multiple projections